File: AmsterdamEats.Maui/AmsterdamEats.Maui/ViewModels/BookingViewModel.cs
using CommunityToolkit.Mvvm.ComponentModel;
using AmsterdamEats.Models;

namespace AmsterdamEats.ViewModels;

public partial class BookingViewModel : ObservableObject
{
    [ObservableProperty]
    private BookingRequest? _bookingRequest;

    public void Prepare(Restaurant restaurant, SearchCriteria criteria)
    {
        BookingRequest = new BookingRequest
        {
            ZenchefRestaurantId = restaurant.ZenchefRestaurantId ?? 0,
            Date = criteria.DateString,
            Pax = criteria.NumberOfPeople,
            RestaurantName = restaurant.Name
        };
    }
}

File: AmsterdamEats/AmsterdamEats/ViewModels/BookingViewModel.swift
import Foundation

@MainActor
final class BookingViewModel: ObservableObject {
    @Published var loadFailed = false
    let bookingRequest: BookingRequest

    init(restaurant: Restaurant, criteria: SearchCriteria) {
        let dateString = DateFormatter.zenchefDate.string(from: criteria.date)
        bookingRequest = BookingRequest(
            zenchefRestaurantId: restaurant.zenchefRestaurantId ?? 0,
            date: dateString,
            pax: criteria.numberOfPeople,
            restaurantName: restaurant.name
        )
    }
}
